server/src/packages/rwcds-form/rwcds-form.service.ts

```diff
@@ -172,13 +177,26 @@ export class RwcdsFormService {
 
       // Lookup: dcp_affectedzoningresolutions asks for a ZR type,
       const { label, code } = ZONING_RESOLUTION_TYPES.filter((zr) => zr.label === projectActionLabel)[0] || { label: '', code: null };
+      const matchingZr = affectedZoningResolutions.find(zr => zr.dcp_zoningresolutiontype === code);
+      // action attribute that matches dcp_zrsectionnumber on dcp_affectedzoningresolution
+      const currentActionZoningResolution = action.dcp_ZoningResolution ? action.dcp_ZoningResolution.dcp_zoningresolution : null
 
       if (!label) console.log(`Could not find Affected ZR Type for ${projectActionLabel}`);
 
+      // if the action.ZoningResolution.dcp_zoningresolution value has changed since the initial POST,
+      // then PATCH the new value to dcp_zrsectionnumber on dcp_affectedzoningresolution entity
+      // NOTE: dcp_modifiedzrsectionnumber not included here because it's updated by applicant on rwcds-form
+      if (zrTypes.includes(label) && label && matchingZr.dcp_zrsectionnumber !== currentActionZoningResolution) {
```

Comment:
Here, can't we just check that `matchingZr` exists?
```
if(matchingZr && matchingZr.dcp_zrsectionnumber !== currentActionZoningResolution) {...}
```
Doesn't the existence of `matchingZr` imply that `label` is within `zrTypes`?

Reply:
If there is a matching rwcds affected zoning resolution for the current project action, then the current project action's label should naturally be in the list of all rwcds affected zoning resolutions

Reply:
Yes that's a good point! The use of `zrTypes` and `label` is left over from old logic/code, and just hasn't been changed yet to match new updates. Thanks for catching that :)

Reply:
np! Thanks for confirming. I think this PR looks good after this fix. 🚀